### PR TITLE
Promote CertificateSigningRequest's Spec.ExpirationSeconds field to GA

### PR DESCRIPTION
**Pull request by cfryanr**
#### What type of PR is this?

/kind feature

This PR is in support of moving the feature from #99494 towards GA.

#### What this PR does / why we need it:

Promote CertificateSigningRequest's Spec.ExpirationSeconds field to GA by:
- Remove the comment "As of v1.22, this field is beta and is controlled via the CSRDuration feature gate" from the expirationSeconds field's godoc.
- Mark the "CSRDuration" feature gate as GA in 1.24, lock its value to "true", and remove the various logic which handled when the gate was "false". Add a comment to remove it in v1.26.
- Update conformance test to check that the CertificateSigningRequest's Spec.ExpirationSeconds field is stored, but do not check if the field is honored since this functionality is optional.

See also the requirements for GA in the related KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2784-csr-duration#ga.

/sig auth

#### Special notes for your reviewer:

/cc @enj

#### Does this PR introduce a user-facing change?

```release-note
The CertificateSigningRequest `spec.expirationSeconds` API field has graduated to GA. The `CSRDuration` feature gate for the field is now unconditionally enabled and will be removed in 1.26.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by enj** [comment]
Since this PR cannot be merged until all the GA items are complete, lets go ahead and update

https://github.com/kubernetes/kubernetes/blob/4d08582d1fa21e1f5887e73380001ac827371553/pkg/features/kube_features.go#L724-L728

with `ga: v1.24` and

https://github.com/kubernetes/kubernetes/blob/4d08582d1fa21e1f5887e73380001ac827371553/pkg/features/kube_features.go#L944

to

```go
CSRDuration: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.25
```

@liggitt does locking the gate to `true` and removing in v1.25 sound reasonable to you?

**Comment by liggitt** [comment]
> Since this PR cannot be merged until all the GA items are complete

yeah, I'd expect the ga-only conformance job to bark at this without the feature gate graduation

> @liggitt does locking the gate to true and removing in v1.25 sound reasonable to you?

lock to true, remove in 1.26, but yes

**Comment by liggitt** [comment]
(and also update the API doc on the field to drop the beta / feature gate reference)

**Comment by cfryanr** [comment]
@enj @liggitt I made the updates that you suggested, if I understood them correctly. Please take a look at the diffs when you have a chance.

I was also wondering if now that this PR makes the field GA, does that mean that it should be mentioned in the PR description in the ` ```docs ` section?

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -941,7 +942,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	SeccompDefault:                                 {Default: false, PreRelease: featuregate.Alpha},
 	PodSecurity:                                    {Default: true, PreRelease: featuregate.Beta},
 	ReadWriteOncePod:                               {Default: false, PreRelease: featuregate.Alpha},
-	CSRDuration:                                    {Default: true, PreRelease: featuregate.Beta},
+	CSRDuration:                                    {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.26
```
Comment (enj) [inline pkg/features/kube_features.go:945]: Lets create an issue to track this removal in 1.26.

**Comment by enj** [comment]
> I was also wondering if now that this PR makes the field GA, does that mean that it should be mentioned in the PR description in the `` ```docs `` section?

Yeah we likely want a PR similar to https://github.com/kubernetes/website/pull/28070 that states that the feature went GA in 1.24.

**Comment by enj** [comment]
@cfryanr because the feature gate is locked to true, we need to basically remove all use of it, and remove any tests that try to disable it.

```
 === FAIL: pkg/controller/certificates/signer Test_signer_duration/feature_gate_disabled_-_can_request_shorter_duration_than_TTL (0.00s)
    feature_gate.go:36: error setting CSRDuration=false: cannot set feature gate CSRDuration to false, feature is locked to true
    signer_test.go:424: duration() = 30m0s, want 1h0m0s
W0318 19:42:57.505243   52272 feature_gate.go:237] Setting GA feature gate CSRDuration=true. It will be removed in a future release. 
```

**Comment by cfryanr** [comment]
> @cfryanr because the feature gate is locked to true, we need to basically remove all use of it, and remove any tests that try to disable it.
> 
> ```
>  === FAIL: pkg/controller/certificates/signer Test_signer_duration/feature_gate_disabled_-_can_request_shorter_duration_than_TTL (0.00s)
>     feature_gate.go:36: error setting CSRDuration=false: cannot set feature gate CSRDuration to false, feature is locked to true
>     signer_test.go:424: duration() = 30m0s, want 1h0m0s
> W0318 19:42:57.505243   52272 feature_gate.go:237] Setting GA feature gate CSRDuration=true. It will be removed in a future release. 
> ```

Thanks, yeah I saw the unit test failures. Nice tests. Made the necessary updates to the PR.

File: pkg/features/kube_features.go (hunk 1 above)
Comment (cfryanr) [reply pkg/features/kube_features.go:945]: Created https://github.com/kubernetes/kubernetes/issues/108814.

**Comment by cfryanr** [comment]
> > I was also wondering if now that this PR makes the field GA, does that mean that it should be mentioned in the PR description in the `` ```docs `` section?
> 
> Yeah we likely want a PR similar to [kubernetes/website#28070](https://github.com/kubernetes/website/pull/28070) that states that the feature went GA in 1.24.

@enj I looked at that doc again. That page on the web site never mentioned that the field was beta, and never mentioned that there was a feature gate related to that field. It only talks about the field being new in 1.22, which will still be true after this PR is merged. So it seems like there is nothing to update on that page for this PR? Is that right?

**Comment by cfryanr** [comment]
Looks like all tests passed and this PR is ready for review @enj @liggitt.

Couple of questions:
- Is promoting this field and locking this feature gate something that would go into release notes? I wasn't sure how to fill out the release notes section of the PR description.
- Does this PR require API review? It changes the doc for an API.

Thanks!

**Comment by liggitt** [comment]
yeah, I updated the release note with details.

/lgtm
/approve

**Pull request merged**